### PR TITLE
Update release checklist

### PR DESCRIPTION
**Pull request by icecrime**
Add early API version bump.
Ping @jfrazelle @moxiegirl.

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

**Comment by moxiegirl** [comment]
LGTM

File: project/RELEASE-CHECKLIST.md [hunk 1]
```diff
@@ -49,7 +49,17 @@ git cherry-pick <commit-id>
 ...
 ```
 
-### 2. Update CHANGELOG.md
+### 2. Bump the API version on master
+
+We don't want to stop contributions to master just because we are releasing. At
+the same time, now that the release branch exists, we don't want API changes to
+go the now frozen API version.
```
Comment (thaJeztah) [inline project/RELEASE-CHECKLIST.md]: "go the now" -> "go to the now"

or just "the now"

File: project/RELEASE-CHECKLIST.md [hunk 2]
```diff
@@ -49,7 +49,17 @@ git cherry-pick <commit-id>
...
+
+Create a new entry in `docs/sources/reference/api/` by copying the latest and
+bumping the version number (in both the file's name and content), and submit
+this in a PR against master.
```
Comment (thaJeztah) [inline project/RELEASE-CHECKLIST.md]: I think it also has to be added to [mkdocs.yml](https://github.com/docker/docker/blob/master/docs/mkdocs.yml#L152)

Or not yet at this stage?

Comment (icecrime) [inline project/RELEASE-CHECKLIST.md]: No idea, I'll leave it to docs maintainers!

File: project/RELEASE-CHECKLIST.md (hunk 1 above)
Comment (icecrime) [inline project/RELEASE-CHECKLIST.md]: Thanks, fixed.

File: project/RELEASE-CHECKLIST.md (hunk 2 above)
Comment (thaJeztah) [inline project/RELEASE-CHECKLIST.md]: @moxiegirl ^^ perhaps you know what's best here?

**Pull request merged**